NFC_EEPROM/source/target/TARGET_ST25DV/EEPROMDriver.cpp

```diff
@@ -0,0 +1,24 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "EEPROMDriver.h"
+#include "st25dv_driver.h"
+
+mbed::nfc::NFCEEPROMDriver& get_eeprom_driver(events::EventQueue&)
+{
+    static mbed::nfc::vendor::ST::ST25dvDriver eeprom_driver;
```

Comment:
I looked at the driver, I'm a bit concerned it doesn't use the event queue, I'd have expected the callbacks to be called in an event to limit stack size. 
The change should be trivial: 

```c++
    virtual void start_session(bool force = true) {
        int ret;
        tr_debug("start_session\r\n");
        if(_is_session_started) {
          delegate()->on_session_started(true);
        }

        ret = open_session(force);
        if(ret != 0) {
          delegate()->on_session_started(false);
        } else {
          _is_session_started = true;
          delegate()->on_session_started(true);
        }
    }

// to be transformed into the following code assuming a pointer to 
// the event queue is available


    virtual void start_session(bool force = true) {
        int ret;
        tr_debug("start_session\r\n");
        if(_is_session_started) {
          eq->call([this] { 
              delegate()->on_session_started(true) 
           });
        }

        ret = open_session(force);
        if(ret != 0) {
          eq->call([this] { 
              delegate()->on_session_started(false) 
           });
        } else {
          _is_session_started = true;
          eq->call([this] { 
              delegate()->on_session_started(true) 
           });
        }
    }

```

Reply:
Thanks @pan- for the hint. Applied accordingly.